docs/rebase.md

```diff
@@ -0,0 +1,125 @@
+
+## Rebasing `Makefile`, `Dockerfiles`, and `.spec` file
+When updating to a new minor version of OpenShift, update the `Makefile` [here](https://github.com/openshift/microshift/blob/main/Makefile#L16) and [here](https://github.com/openshift/microshift/blob/main/Makefile#L55-L66) and also check whether the Dockerfiles need a new base image (e.g. [this](https://github.com/openshift/microshift/blob/main/packaging/images/openshift-ci/Dockerfile)) and the RPM `.spec` file needs updates (e.g. [this](https://github.com/openshift/microshift/blob/main/packaging/rpm/microshift.spec))
+
+Commit the changes:
+
+```
+git add vendor
+git commit -m "update Makefile and Dockerfiles"
+```
+
+## Rebasing the embedded components
+### Component images
+To update the component image references of the MicroShift release, run:
+
+```
+./scripts/rebase.sh images
+git add pkg/release
+git commit -m "update component images"
+```
+
+Note that as long as we don't use 100% OpenShift-built images, you may need to bump the `pause` and network and storage provider images manually.
+
+### Component manifests
+The next step is still poorly automated by `rebase.sh` and definitely requires manual review. It basically just gathers the various resource manifests used by the OpenShift control plane and the hosted components. In particular the latter are a few cases just the _templates_ of the manifests used by the hosted component's Operator, i.e. we need to substitute those for example with the variables MicroShift renders into the template or the name, namespace, or labels that the Operator would add in OpenShift.
+
+Gather the manifests / manifest templates into the respective asset directory:
+
+```
+./scripts/rebase.sh manifests
+```
+
+Review the changes. In particular:
+* restore names, namespaces, labels, and annotations that have been replaced with comments like "# X is set at runtime."
+* restore the substitution of placeholders with variables for container images
+* restore the change of the openshift-router's service from type `LoadBalancer` to `NodePort`
+* ensure all `imagePullPolicy`s are set to `IfNotPresent` so offline mode works
+* review where replica counts can be set to 1, e.g. `assets/components/service-ca/deployment.yaml`
+* restore the following Operator-generated assets and compare to OpenShift whether they need updates:
+```
+assets/components/openshift-dns/dns/configmap.yaml
+assets/components/openshift-dns/node-resolver/daemonset.yaml
+assets/components/openshift-router/configmap.yaml
+```
+
+Embed the assets into the binary and commit the result:
+
+```
+./scripts/bindata.sh
+git add assets pkg/assets
+git commit -m "update manifests"
+```
+
+### Component configs
+The last step isn't automated at all yet, which is to compare whether the config parameters of embedded component changed.
```

Comment:
Can we add at least an example of what needs to be compared to what?

Reply:
Added this.